tests/pages/pay-page.js

```diff
@@ -1,16 +1,34 @@
-import { clickable, collection, create, text, triggerable, visitable } from 'ember-cli-page-object';
+import { clickable, collection, create, text, visitable } from 'ember-cli-page-object';
 import AccountDropdown from 'codecrafters-frontend/tests/pages/components/account-dropdown';
 import Header from 'codecrafters-frontend/tests/pages/components/header';
 
 export default create({
   accountDropdown: AccountDropdown,
-  clickOnApplyRegionalDiscountButton: clickable('[data-test-apply-regional-discount-button]'),
+
+  clickOnChoosePlanButton: clickable('[data-test-choose-plan-button]'),
   clickOnExtraInvoiceDetailsToggle: clickable('[data-test-extra-invoice-details-toggle]'),
   clickOnProceedToCheckoutButton: clickable('[data-test-proceed-to-checkout-button]'),
-  clickOnStartPaymentButtonForMonthlyPlan: clickable('[data-test-monthly-pricing-card] [data-test-start-payment-button]'),
-  clickOnStartPaymentButtonForYearlyPlan: clickable('[data-test-yearly-pricing-card] [data-test-start-payment-button]'),
+
   header: Header,
 
+  pricingPlanCards: collection('[data-test-pricing-plan-card]', {
+    ctaButton: {
+      scope: '[data-test-pricing-plan-card-cta]',
+    },
+  }),
+
+  chooseMembershipPlanModal: {
+    planCards: collection('[data-test-modal-plan-card]', {
```

Comment:
```suggestion
    planCards: collection('[data-test-plan-card]', {
```

Add `scope: 'data-test-choose-membership-plan-modal'` to this and we won't have to keep qualifying each `data-test` property as `data-test-modal` (it'll automatically use nested to only find out relevant items